libcxx/test/libcxx/language.support/nodiscard.verify.cpp

```diff
@@ -0,0 +1,91 @@
+    std::is_lteq(oRes); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+    std::is_gt(oRes);   // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+    std::is_gteq(oRes); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+  }
+#endif
+
+#if TEST_STD_VER >= 20
+  { // <coroutine>
+    struct EmptyPromise {
+    } promise;
+
+    {
+      std::coroutine_handle<void> cr{};
+
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      cr.address();
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      std::coroutine_handle<void>::from_address(&promise);
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      cr.done();
+
+      std::hash<std::coroutine_handle<void>> hash;
+      hash(cr); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+    }
+    {
+      std::coroutine_handle<EmptyPromise> cr;
+
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      std::coroutine_handle<EmptyPromise>::from_promise(promise);
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      cr.address();
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      std::coroutine_handle<EmptyPromise>::from_address(&promise);
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      cr.done();
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      cr.promise();
+    }
+    {
+      std::coroutine_handle<std::noop_coroutine_promise> cr = std::noop_coroutine();
+
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      cr.done();
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      cr.promise();
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      cr.address();
+
+      // expected-warning@+1 {{ignoring return value of function declared with 'nodiscard' attribute}}
+      std::noop_coroutine();
+    }
+  }
+#endif
+
+  { // <initializer_list>
+    std::initializer_list<int> il{94, 82, 49};
+
+    il.size();  // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+    il.begin(); // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
+    il.end();   // expected-warning {{ignoring return value of function declared with 'nodiscard' attribute}}
```

Comment:
Non-member `begin`/`end` were not changed. But IMO this is good, because they should be removed in all modes (by [P3016R6](https://wg21.link/p3016r6)) and we won't need to bother to delete lines here.